src/app/request-handlers/guards/survey-participation-guard.ts

```diff
@@ -14,7 +14,7 @@ export class SurveyParticipationGuard extends RequestHandlerGuard {
     public async check(onReject: (message: string) => Promise<void>): Promise<void> {
         if (this.participationId == '') {
             onReject(
-                'Token ATM cannot access your email address, which thus cannot check your participation. Please make your email address visible.'
+                'Token ATM cannot access your email address, and so cannot check your participation. Please make your email address visible.'
```

Comment:
You're right that this error message needs to be enhanced. It should be on the Canvas side. On Qualtrics side, students are logged in using Shibboleth SSO, so they have no control over their email address availability. On Canvas, according to [a comnunity post](https://community.canvaslms.com/t5/Canvas-Question-Forum/How-do-I-hide-my-email-address/m-p/216078), the visibility of email address is the institution's decision, so they also have no control over that. 

However, a student could choose to add another email to Canvas and select that email as the default email address. Since all Canvas API returns the default email address, this will cause the mismatch of Canvas email address and Qualtrics email address. 

Overall, the error message here should be something like `Your Canvas email address is not visible, please contact your institution for help`, but I don't think such a decision of hiding email address will be easily changed. In addition, the error message on line 27 should indicate the possibility of customized default email address and inform the student to change the default email address back to the original one to keep the record match with Qualtrics.